OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -811,13 +837,23 @@ function Install-OctopusDeploy {
     Update-InstallState "OctopusAdminUsername" $extractedUsername
     Update-InstallState "OctopusAdminPassword" ($OctopusAdminCredential.Password | ConvertFrom-SecureString)
 
-    Write-Log "Configuring Octopus Deploy instance to use free license ..."
-    $args = @(
-        'license',
-        '--console',
-        '--instance', $name,
-        '--free'
-    )
+    if (($null -eq $licenseKey) -or ($licenseKey -eq "")) {
```

Comment:
I'd maybe refactor this bit a little to make it more compact, but that's really just a style consideration. We can do that later.

```
if($null -ne $licenceKey)
{
$args += @('--licencekeyBase64', $licenceKey)
}
else
{
$args += @('--free')
}
```

Reply:
Yeah, good call.